Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1990,6 +1990,12 @@ std::tuple<Ref<WebProcessProxy>, SuspendedPageProxy*, ASCIILiteral> WebProcessPo
     if (m_automationSession)
         return { WTFMove(sourceProcess), nullptr, "An automation session is active"_s };
 
+    // Redirects to a different protocol for which we have a custom scheme handler registered need to
+    // process swap, so that we end up with a fresh navigation to the custom scheme and so that the
+    // custom protocol handler is used (rdar://117891282).
+    if (navigation.currentRequestIsRedirect() && navigation.originalRequest().url().protocol() != targetURL.protocol() && page.urlSchemeHandlerForScheme(targetURL.protocol().toString()))
+        return { createNewProcess(), nullptr, "Redirect to a custom protocol"_s };
```

Comment:
We don't know that it's a *custom* protocol here, right? Just that it's a different protocol. We should change the string to reflect that.

Reply:
We called page.urlSchemeHandlerForScheme() and it returned non-null, so it is a custom protocol that the app registered a handler for.

Reply:
It is not just "any" different protocol.

Reply:
I don't see where the dependency on custom protocol comes in.  Is it `page.urlSchemeHandlerForScheme()` that scopes this to only custom protocols? That sounds implicit and I would change the string to match the naming of that function more closely.

Reply:
Maybe I'm using the wrong term? I used "custom protocol" to mean a "scheme for which the client registered their own handler". Maybe I can change the logging to use that instead, because that's exactly what I'm checking for.

Reply:
Yeah, that sounds much better.